src/platform/backends/hyperv_api/hcs_virtual_machine_factory.cpp

```diff
@@ -0,0 +1,291 @@
+                      origin_interface_name,
+                      extra.id);
+            continue;
+        }
+
+        const auto vswitch_name = create_bridge_with(found->second);
+        if (vswitch_name.empty())
+        {
+            mpl::warn(log_category, "vSwitch {} could not be created", found->first);
+        }
+    }
+
+    return std::make_unique<HCSVirtualMachine>(default_hyperv_switch_guid,
+                                               desc,
+                                               monitor,
+                                               key_provider,
+                                               get_instance_directory(desc.vm_name));
+}
+
+void HCSVirtualMachineFactory::remove_resources_for_impl(const std::string& name)
+{
+    mpl::debug(log_category, "remove_resources_for_impl() -> VM: {}", name);
+    hcs::HcsSystemHandle handle{nullptr};
+    if (HCS().open_compute_system(name, handle))
+    {
+        // Everything for the VM is neatly packed into the VM folder, so it's enough to ensure that
+        // the VM is stopped. The base class will take care of the nuking the VM folder.
+        const auto& [status, status_msg] = HCS().terminate_compute_system(handle);
+        if (status)
+        {
+            mpl::warn(log_category,
+                      "remove_resources_for_impl() -> Host compute system {} was still alive.",
+                      name);
+        }
+    }
+    else
+    {
+        mpl::info(log_category,
+                  "remove_resources_for_impl() -> Host compute system `{}` already terminated.",
+                  name);
+    }
+}
+
+VMImage HCSVirtualMachineFactory::prepare_source_image(const VMImage& source_image)
+{
+    const std::filesystem::path source_file{source_image.image_path.toStdString()};
+
+    if (!std::filesystem::exists(source_file))
+    {
+        throw ImageConversionException{"Image {} does not exist", source_file};
+    }
+
+    const std::filesystem::path target_file = [source_file]() {
+        auto target_file = source_file;
+        target_file.replace_extension(".vhdx");
+        return target_file;
+    }();
+
+    const QStringList qemu_img_args{"convert",
+                                    "-o",
+                                    "subformat=dynamic",
+                                    "-O",
+                                    "vhdx",
+                                    QString::fromStdString(source_file.string()),
+                                    QString::fromStdString(target_file.string())};
+
+    QProcess qemu_img_process{};
+    qemu_img_process.setProgram("qemu-img.exe");
+    qemu_img_process.setArguments(qemu_img_args);
+    qemu_img_process.start();
+
+    if (!qemu_img_process.waitForFinished(multipass::image_resize_timeout))
+    {
+        throw ImageConversionException{"Conversion of image {} to VHDX timed out", source_file};
+    }
+
+    if (qemu_img_process.exitCode() != 0)
```

Comment:
[`QProcess::exitCode`](https://doc.qt.io/qt-6/qprocess.html#exitCode) says:

> This value is not valid unless [exitStatus](https://doc.qt.io/qt-6/qprocess.html#exitStatus)() returns [NormalExit](https://doc.qt.io/qt-6/qprocess.html#ExitStatus-enum).

Which is why the old Hyper-V implementation used:

```
    if (convert.exitStatus() != QProcess::NormalExit || convert.exitCode() != 0)
```

I believe we should match that here.

Reply:
done